syntax = "proto2";

import "protos/preprocess.proto";
import "protos/image_resizer.proto";

message Reader {
  oneof reader_oneof {
    Cap2DetReader cap2det_reader = 1;
  }
}

message Cap2DetReader {
  // Pattern of the input files.
  repeated string input_pattern = 1;

  // Cycle length of interleave call.
  optional int32 interleave_cycle_length = 2 [default = 2];

  // If True, repeat the dataset and shuffle the batch.
  optional bool is_training = 3 [default = false];

  // Shuffle buffer size.
  optional int32 shuffle_buffer_size = 4 [default = 1000];

  // Number of parallel calls.
  optional int32 map_num_parallel_calls = 5 [default = 1];

  // Prefetch buffer size.
  optional int32 prefetch_buffer_size = 6 [default = 200];

  // Batch size.
  optional int32 batch_size = 7 [default = 32];

  // If false, do not decode image; For text classifier training.
  optional bool decode_image = 11 [default = true];

  // Image resizer.
  optional ImageResizer image_resizer = 12;

  // Preprocess options.
  optional Preprocess preprocess_options = 13;

  // Pad to the max number of proposals to ensure static shape.
  optional int32 max_num_proposals = 14 [default = 500];

  // Randomly resize image according to these scales values.
  repeated float batch_resize_scale_value = 15;

  // Shard to read, in the format of '0/3', '1/3', '2/3'.
  // The denominator denotes the number of shards. The numerator denotes the shard to export.
  optional string shard_indicator = 16;
}
